### PR TITLE
determine and print the agent version by default

### PR DESCRIPTION
**Pull request by rPraml**
As we use a lot of run-time enhancement (and not the eclipse plugin) in development it is neccessary to enhance with the correct agent version.

BTW: we configure enhancement by adding `-javaagent:YOUR_HOME_DIR/.m2/repository/io/ebean/ebean-agent/11.xx.y/ebean-agent-11.xx.y.jar=debug=0` as default VM arguments in eclipse.

Changes in this PR:

*Add a static `getVersion()` in io.ebean.enhance.Transformer*
This method can be invoked by the application via reflection (see example below) to check if the agent version matches with the version defined in the POM file (the enhancer used for production)

*Print current agent version at startup* 
This is just for information and should help for debugging enhancer issues (in goolge groups etc.)

Reflection Example:

```java
public static final String REQUIRED_VERSION = "${ebean-agent.version}" // replace by maven resource filter
public static void check() {
	try {
		Class<?> cls = Class.forName("io.ebean.enhance.Transformer");
		try {
			Method method = cls.getMethod("getVersion");
			String version = (String) method.invoke(null);
			if (!REQUIRED_VERSION.equals(version)) {
				throw new IllegalStateException("Wrong enhancer version");
			}
		} catch (Exception e) {
			throw new IllegalStateException("Could not detect enhancer version");
		}
	} catch (ClassNotFoundException e) {
		// no enhancer present - does not matter
	}
}
```
This is executed in our application after start up (maybe we can integrate that in ebean-server initialization to ensure, that ebean always uses the correct agent)

### CODE REVIEW

File: src/main/java/io/ebean/enhance/Transformer.java [hunk 1]
```diff
@@ -37,6 +40,25 @@
  */
 public class Transformer implements ClassFileTransformer {
 
+  private static String version = "unknown";
+  static {
```
Comment (rbygrave) [inline src/main/java/io/ebean/enhance/Transformer.java]: I don't think we should do this on cinit (all the time) but instead have it as a method that when invoked reads the pom.properties file.

File: src/main/java/io/ebean/enhance/common/EnhanceContext.java [hunk 2]
```diff
@@ -95,6 +96,9 @@ public EnhanceContext(ClassBytesReader classBytesReader, String agentArgs, Agent
 
     this.transientInternalFields = getPropertyBoolean("transientInternalFields", manifest.isTransientInternalFields());
     this.checkNullManyFields = getPropertyBoolean("checkNullManyFields", manifest.isCheckNullManyFields());
+    if (getPropertyBoolean("printversion", true)) {
```
Comment (rbygrave) [inline src/main/java/io/ebean/enhance/common/EnhanceContext.java]: I don't think we should bother with this but leave it up to application code ?

File: src/main/java/io/ebean/enhance/Transformer.java (hunk 1 above)
Comment (rPraml) [reply src/main/java/io/ebean/enhance/Transformer.java]: I agree

File: src/main/java/io/ebean/enhance/common/EnhanceContext.java (hunk 2 above)
Comment (rPraml) [reply src/main/java/io/ebean/enhance/common/EnhanceContext.java]: How do you mean that? The application code (=code that gets enhanced at runtume) does not know, anything about the agent. 

So I thought, if run time enhancement is used and someone do not want that the agent prints the version, he/she can add `-javaagent:ebean-agent-${version}.jar=printVersion=false` as agent argument.

Comment (rbygrave) [reply src/main/java/io/ebean/enhance/common/EnhanceContext.java]: Sorry, I wasn't clear there. I was thinking just default it to `false` ... so it only prints to sysout when explicitly requested.  

Thinking that framework code would use the static `Transformer.getVersion()` ... might be rare that we want the version in sysout.

**Pull request merged**